### ISSUE TITLE
Force HTTPS redirect on heroku

### ISSUE THREAD

**Issue by commjoen** [commjoen/wrongsecrets#157]
During our webinar with GitGuardian, we demoed the app , by mistake using http at heroku, which led to an issue with the cookies, not keeping our progress.

Therefore we need to make sure that at Heroku we keep redirecting http traffic to https, with for instance, https://devcenter.heroku.com/articles/preparing-a-spring-boot-app-for-production-on-heroku#force-the-use-of-https

**Comment by nbaars** [comment]
Can't you use the the proposed example from Heroku? Seems a valid configuration and does not interfere with running it locally:

```
@Configuration
public class WebSecurityConfig extends WebSecurityConfigurerAdapter {

  @Override
  protected void configure(HttpSecurity http) throws Exception {
    http.requiresChannel()
      .requestMatchers(r -> r.getHeader("X-Forwarded-Proto") != null)
      .requiresSecure();
  }
}
```

**Comment by commjoen** [comment]
Yes that is the one I meant with https://devcenter.heroku.com/articles/preparing-a-spring-boot-app-for-production-on-heroku#force-the-use-of-https but I thought it would cause some interference on some of our deployment variants

**Comment by commjoen** [comment]
hmm it seems i need to have another version of spring security for this.. let me check

**Comment by commjoen** [comment]
@nbaars this solution will put on spring cloud security with auth :(

**Comment by commjoen** [comment]
Fixed with #173
